mucommander-os-api/src/main/java/com/mucommander/desktop/DefaultDesktopAdapter.java

```diff
@@ -155,4 +157,54 @@ public boolean isApplication(AbstractFile file) {
     public ActionShortcuts getActionShortcuts() {
         return new ActionShortcuts();
     }
+
+    @Override
+    public boolean setIconBadgeNumber(int numberOfJobs) {
+        boolean opSupportedByAwt = false;
+        if (Taskbar.isTaskbarSupported() &&
+                Taskbar.getTaskbar().isSupported(Taskbar.Feature.ICON_BADGE_TEXT)) {
+            if (numberOfJobs > 0) {
+                Taskbar.getTaskbar().setIconBadge(Integer.toString(numberOfJobs));
+            } else {
+                Taskbar.getTaskbar().setIconBadge(null);
+            }
+            opSupportedByAwt = true;
+        }
+        return opSupportedByAwt;
+    }
+
+    @Override
+    public boolean setIconProgress(int progress) {
+        boolean opSupportedByAwt = false;
+        if (Taskbar.isTaskbarSupported() &&
+                Taskbar.getTaskbar().isSupported(Taskbar.Feature.PROGRESS_VALUE)) {
+            if (progress >= 0 && progress <= 100) {
+                Taskbar.getTaskbar().setProgressValue(progress);
+            } else {
+                Taskbar.getTaskbar().setProgressValue(-1);
+            }
```

Comment:
@pskowronek what I meant before is whether we can move the 5 lines above to the place that is invoked later below

Reply:
ok, PR updated